executor/adapter.go

```diff
@@ -63,6 +63,22 @@ func (a *statement) IsDDL() bool {
 }
 
 func (a *statement) Exec(ctx context.Context) (ast.RecordSet, error) {
+	if execPlan, ok := a.plan.(*plan.Execute); ok {
+		// For execute plan, we need to get the underlying prepared statement node.
+		prepared, _ := getPreparedStmt(ctx, execPlan.Name, execPlan.ID)
+		if prepared == nil {
+			return nil, ErrStmtNotFound
+		}
+		a.node = prepared.Stmt
+	}
+
+	// Execute non-correlated subquery before build executor.
+	subExec := &subqueryExecutor{ctx: ctx}
+	a.node.Accept(subExec)
```

Comment:
If we use an index to get handles and use subquery as filter, when get empty set from index, it is not necessary to evaluate subquery.